app/src/main/java/com/asfoundation/wallet/repository/ProdTransactionRepository.kt

```diff
@@ -0,0 +1,85 @@
+package com.asfoundation.wallet.repository
+
+import com.asf.wallet.BuildConfig
+import com.asfoundation.wallet.entity.NetworkInfo
+import com.asfoundation.wallet.entity.RawTransaction
+import com.asfoundation.wallet.entity.Wallet
+import com.asfoundation.wallet.interact.DefaultTokenProvider
+import com.asfoundation.wallet.poa.BlockchainErrorMapper
+import com.asfoundation.wallet.service.AccountKeystoreService
+import com.asfoundation.wallet.service.TransactionsNetworkClientType
+import com.asfoundation.wallet.transactions.Transaction
+import com.asfoundation.wallet.transactions.TransactionsMapper
+import com.asfoundation.wallet.ui.iab.raiden.MultiWalletNonceObtainer
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.Single
+
+class ProdTransactionRepository(
+    private val networkInfo: NetworkInfo,
+    accountKeystoreService: AccountKeystoreService,
+    private val inDiskCache: TransactionLocalSource,
+    private val blockExplorerClient: TransactionsNetworkClientType,
+    defaultTokenProvider: DefaultTokenProvider,
+    errorMapper: BlockchainErrorMapper,
+    nonceObtainer: MultiWalletNonceObtainer,
+    scheduler: Scheduler,
+    private val mapper: TransactionsMapper,
+    private val offChainTransactions: OffChainTransactions
+) : TransactionRepository(networkInfo, accountKeystoreService,
+    defaultTokenProvider, errorMapper, nonceObtainer, scheduler) {
+
+  override fun fetchTransaction(wallet: String): Observable<List<Transaction>> {
+    return Observable.merge(getOnchainTransactions(networkInfo, wallet),
```

Comment:
I think networkInfo should not be passed as an argument to this method since it’s a field of the class